### DISCUSSION TITLE
Add rotation to axis values' labels

### DISCUSSION THREAD

**Discussion by GuilhE** [patrykandpatrick/vico#673] [Questions] [answered]
Is it possible to add rotation to axis values' labels to achieve something like:

![Screenshot 2024-04-26 at 09 45 47](https://github.com/patrykandpatrick/vico/assets/2677139/30d172b3-5db9-40ab-901e-311343b29c3f)

I can only make it vertical with `labelRotationDegrees = -90`:

![Screenshot 2024-04-26 at 09 47 09](https://github.com/patrykandpatrick/vico/assets/2677139/21535065-28db-495b-9b76-703d24998661)

Because if I add `labelRotationDegrees = -45` I get:

![Screenshot 2024-04-26 at 09 55 26](https://github.com/patrykandpatrick/vico/assets/2677139/28e842c7-14ae-4da1-89c0-fc6da5b7500c)

If I change `lineCount = 10`:

![Screenshot 2024-04-26 at 09 59 05](https://github.com/patrykandpatrick/vico/assets/2677139/7f0270bd-8d49-4ec0-80d8-2592cccb1013)

**Comment by Gowsky** [comment, marked as answer]
Hello, @GuilhE. You can apply any rotation angle, but currently, labels aren’t allowed to extend past their segments:

<img width="256" height="108" alt="image" src="https://github.com/user-attachments/assets/fd32ed8a-69ad-4b26-b0d2-a6b34c870a22" />

That’s the reason for the behavior you’re seeing. The labels have nearly no room and are truncated.

The aforementioned width restriction works for most rotations (most notably, 0 and 90 degrees), but it’s excessive in certain 45-degree setups (such as yours), where labels wouldn’t overlap even if their extended past their segments. Feel free to open a feature request for this to be improved.
